src/statement/baseStatement.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  ExecuteQueryOptions,
+  Context,
+  StreamOptions,
+  Statistics
+} from "../types";
+import { Meta } from "../meta";
+
+export abstract class BaseStatement {
```

Comment:
Do we ever use this class. Since we have two separate methods for async and sync execution, can we remove this class?

Reply:
Fair point, it was leftover from previous iterations